File: server/routes/skills.js
const express = require('express');
const User = require('../models/User');
const auth = require('../middleware/auth');

const router = express.Router();

// Get user's skills
router.get('/my-skills', auth, async (req, res) => {
  try {
    const user = await User.findById(req.user._id);
    const skills = user.profile?.skills || [];
    
    // Transform skills into detailed format
    const detailedSkills = skills.map(skill => ({
      name: skill,
      level: 'Intermediate', // Default level
      endorsements: Math.floor(Math.random() * 15), // Mock endorsements
      verified: Math.random() > 0.7, // Random verification status
      endorsers: ['Sample User 1', 'Sample User 2'] // Mock endorsers
    }));

    res.json(detailedSkills);
  } catch (error) {
    console.error('Skills fetch error:', error);
    res.status(500).json({ message: 'Server error' });
  }
});

// Add new skill
router.post('/add', auth, async (req, res) => {
  try {
    const { skill, level } = req.body;
    
    const user = await User.findById(req.user._id);
    if (!user.profile.skills) {
      user.profile.skills = [];
    }
    
    if (!user.profile.skills.includes(skill)) {
      user.profile.skills.push(skill);
      await user.save();
    }

    const newSkill = {
      name: skill,
      level: level || 'Beginner',
      endorsements: 0,
      verified: false,
      endorsers: []
    };

    res.json(newSkill);
  } catch (error) {
    console.error('Add skill error:', error);
    res.status(500).json({ message: 'Server error' });
  }
});

// Get pending endorsement requests
router.get('/pending-endorsements', auth, async (req, res) => {
  try {
    // Mock pending endorsements for demo
    const pendingEndorsements = [
      {
        id: 1,
        requester: { 
          name: 'John Doe', 
          email: 'john@example.com' 
        },
        skill: 'React.js',
        message: 'Please endorse my React.js skills based on our recent project collaboration.'
      }
    ];

    res.json(pendingEndorsements);
  } catch (error) {
    console.error('Pending endorsements error:', error);
    res.status(500).json({ message: 'Server error' });
  }
});

// Endorse a skill
router.post('/endorse/:endorsementId', auth, async (req, res) => {
  try {
    const { endorsementId } = req.params;
    const { approve } = req.body;

    // Mock endorsement processing
    console.log(`Endorsement ${endorsementId} ${approve ? 'approved' : 'declined'} by user ${req.user._id}`);

    res.json({ message: 'Endorsement processed successfully' });
  } catch (error) {
    console.error('Endorsement error:', error);
    res.status(500).json({ message: 'Server error' });
  }
});

// Request skill verification
router.post('/request-verification', auth, async (req, res) => {
  try {
    const { skill } = req.body;
    
    // Mock verification request
    console.log(`Verification requested for skill: ${skill} by user ${req.user._id}`);

    res.json({ message: 'Verification request sent to your connections' });
  } catch (error) {
    console.error('Verification request error:', error);
    res.status(500).json({ message: 'Server error' });
  }
});

module.exports = router;

File: server-python/routes/skills.py
from flask import Blueprint, request, jsonify
from models import User
from extensions import db
from routes.middleware import token_required
import random

skills_bp = Blueprint('skills', __name__)

@skills_bp.route('/my-skills', methods=['GET'])
@token_required
def get_my_skills(current_user):
    try:
        profile = current_user.get_profile()
        skills = profile.get('skills', [])
        
        detailed_skills = []
        for skill in skills:
            detailed_skills.append({
                "name": skill,
                "level": 'Intermediate',
                "endorsements": random.randint(0, 15),
                "verified": random.random() > 0.7,
                "endorsers": ['Sample User 1', 'Sample User 2']
            })
            
        return jsonify(detailed_skills), 200
    except Exception as e:
        return jsonify({"message": "Server error", "error": str(e)}), 500

@skills_bp.route('/add', methods=['POST'])
@token_required
def add_skill(current_user):
    try:
        data = request.json or {}
        skill = data.get('skill')
        level = data.get('level', 'Beginner')
        
        if not skill:
            return jsonify({"message": "Skill is required"}), 400
            
        profile = current_user.get_profile()
        if 'skills' not in profile:
            profile['skills'] = []
            
        if skill not in profile['skills']:
            profile['skills'].append(skill)
            current_user.set_profile(profile)
            db.session.commit()
            
        return jsonify({
            "name": skill,
            "level": level,
            "endorsements": 0,
            "verified": False,
            "endorsers": []
        }), 200
    except Exception as e:
        return jsonify({"message": "Server error", "error": str(e)}), 500

@skills_bp.route('/pending-endorsements', methods=['GET'])
@token_required
def get_pending_endorsements(current_user):
    # Mock data
    pending = [
        {
            "id": 1,
            "requester": {"name": "John Doe", "email": "john@example.com"},
            "skill": "React.js",
            "message": "Please endorse my React.js skills"
        }
    ]
    return jsonify(pending), 200

@skills_bp.route('/endorse/<int:endorsement_id>', methods=['POST'])
@token_required
def endorse_skill(current_user, endorsement_id):
    return jsonify({"message": "Endorsement processed successfully"}), 200

@skills_bp.route('/request-verification', methods=['POST'])
@token_required
def request_verification(current_user):
    return jsonify({"message": "Verification request sent to your connections"}), 200
